### PR TITLE
Feature/js 19 300 react/login

### PR DESCRIPTION
**Pull request by karbik-creator**

### CODE REVIEW

File: react/src/features/auth/components/Form.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import { useFormik } from 'formik';
+import * as yup from 'yup';
+import { Link } from 'react-router-dom';
+import { FormHelperText } from '@mui/material';
+
+interface Props {
+  /** Form name . */
+  readonly title: string;
+  /** Event handler on form submit. */
+  readonly handleSubmit: Function;
```
Comment (golovachev-yaroslav) [inline react/src/features/auth/components/Form.tsx:17]: It would be better to describe the function
Ex:
`readonly handleSubmit: (data: SomeType) => void;`

**Review by golovachev-yaroslav — APPROVED** [review]

File: react/src/features/auth/components/Form.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    .required('Password is required'),
+});
+const FormComponent: VFC<Props> = ({
+  title,
+  redirectLinkName,
+  redirectLink,
+  handleSubmit,
+  error,
+}) => {
+  const [open] = React.useState(true);
```
Comment (ShestakovaElena) [inline react/src/features/auth/components/Form.tsx:50]: It seems strange for me - is Dialog window always open?

File: react/src/store/user/state.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+export interface UserState {
+  /** User. */
+  readonly user: User | null;
+  /** Is the user authorized. */
+  readonly isAuthenticated: boolean;
+  /** Authorization error. */
+  readonly error?: string;
+}
+export const initialUserState: UserState = {
+  user: AuthService.getCurrentUser(),
```
Comment (pavelzadorin-saritasa) [inline react/src/store/user/state.ts:16]: the Redux main use case is when the initial state is a static object, essentially state default values.
getting some information (even synchronously) is better done with an explicit action, for example: `fetchCurrentUser` which will get the current user from the Firebase and set it in the store.

**Review by pavelzadorin-saritasa — CHANGES_REQUESTED** [review]

File: react/src/store/user/selectors.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { createSelector } from '@reduxjs/toolkit';
+
+import { RootState } from '../store';
+
+/** Selects user from store. */
+export const selectUser = createSelector((state: RootState) => state.user.user, user => user);
```
Comment (pavelzadorin-saritasa) [inline react/src/store/user/selectors.ts:6]: let's keep the consistency, in all other selectors you refer to `state.user` and then you get a specific parameter (like `user.isAuthenticated`), but here you immediately get the parameter. For consistency, it is better to do as in the rest.

**Review by pavelzadorin-saritasa — APPROVED** [review]

**Review by kristinaSvobodinaSaritasa — APPROVED** [review]

**Pull request merged**